Gemfile.lock

```diff
@@ -21,6 +24,7 @@ GEM
       diff-lcs (>= 1.2.0, < 2.0)
       rspec-support (~> 3.11.0)
     rspec-support (3.11.1)
+    stringio (3.1.0)
```

Comment:
This is resulting in build failures in [PBB ](https://buildkite.com/gusto/payroll-building-blocks/builds/22857#01901297-8937-4493-b149-d4d7d616b292)

```
/usr/local/lib/ruby/3.3.0/bundler/runtime.rb:304:in `check_for_activated_spec!': You have already activated stringio 3.1.0, but your Gemfile requires stringio 3.1.1. Since stringio is a default gem, you can either remove your dependency on it or try updating to a newer version of bundler that supports stringio as a default gem. (Gem::LoadError)
```

Reply:
The main reason here is to remove the YAML patch in https://github.com/rubyatscale/code_manifest/pull/5/files#diff-2aac61581ea2d39514c7db56a64c654b251315a2c74ec6e251e41ba75acabff4L48-L52 which was built-in in psych 4. 

The problem here is that psych puts default gem as dependency: https://github.com/ruby/psych/pull/525 Also from a gem's perspective, I don't think this change on `code_manifest` is inappropriate since we do want to only support psych >= 4.0.

I think the better way to tackle this is to make sure a `gem update --system` ran before `bundle install`.